src/dune_rules/ocaml_flags.ml

```diff
@@ -84,6 +84,13 @@ module Spec = struct
     and+ native = field_oslu "ocamlopt_flags" in
     let specific = Mode.Dict.make ~native ~byte in
     { common; specific }
+
+  let decode_compile_flags_field field_name =
```

Comment:
I think this function should be defined near where it's called. I only see it being used in one place.

Reply:
right

Reply:
If you can find a way to generalize this function to use it in more than one place, that's certainly preferable. Let's just not mix up refactoring and feature development. Therefore, such an improvement should come in a separate PR.

Reply:
Note that the type of `Ocaml_flags.Spec.t` is opaque. So just moving this function near the site of use is not really an option without worse compromises (needless exposure of the value constructors, making the API of `Spec` more brittle). It seem then that the only real option here is to open a separate PR that generalizes the `Ocaml-flags.Spec.decode` function, and make that a prerequisite for this PR.

Reply:
Wait, why don't we just use the existing `Ocaml_flags.Spec.decode` function then.

This seems much more natural:

```
(inline_tests
 (backend ..)
 (exe
  (flags ..)))
```

This way we support both bytecode and native flags. And we don't have make up any new fields.

@lubegasimon you can try this improvement if you'd like. Otherwise, I can give it a shot.

Reply:
Ah -- sure, if the requirements change, we can reuse the existing function. What is `exe` meant to express in the language? That these are the flags that should be passed when compiling the `inline_test` executable? Do we already use this field name somewhere? If not, perhaps instead of `exe` we should call the parent field `compilation` or something a little less overloaded?

Reply:
Perhaps we should call the field `executable` then, and then document that this supports a subset of the fields allowed in the `executable` stanza?

Reply:
> @lubegasimon you can try this improvement if you'd like. Otherwise, I can give it a shot.
cool, no problem, I'll do it. @rgrinberg

Reply:
> (inline_tests
 (backend ..)
 (exe
  (flags ..)))

@rgrinberg, won't another `flags` field bring about confusion because we already have a one optional `flags` field for building a test executable https://dune.readthedocs.io/en/stable/tests.html#passing-special-arguments-to-the-test-runner.

Reply:
That's why I like putting it under `(executable ..)` The following doesn't seem confusing to me:

```
(inline_tests
 (flags ..)
 (executable
  (flags ..)))
```

Although I can see that the other `flags` field would have been better named as `runtime_flags` or something.